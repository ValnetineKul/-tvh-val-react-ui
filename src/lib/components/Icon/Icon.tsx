import React from 'react';
import SvgIcon from '@mui/material/SvgIcon';
import { capitalize } from '@mui/material/utils';
import useStyles from './Icon.styles';

export interface IconProps extends React.AriaAttributes {
  icon: React.FunctionComponent;
  size?: 'md' | 'sm' | 'xl';
  className?: string;
  title?: string;
  isMultiColor?: boolean;
}

const Icon = ({ icon, size = 'md', className, title, isMultiColor = false, ...props }: IconProps) => {
  const { classes, cx } = useStyles();
  const sizeClassName = (classes as Record<string, string>)[`size${capitalize(size)}`];

  const IconComponent = typeof icon === 'string' ? React.lazy(() => import(`./icons/${icon}`)) : icon;

  return (
    <div className={cx(classes.root, sizeClassName, className)}>
      <SvgIcon
        htmlColor={isMultiColor ? 'transparent' : 'inherit'}
        component={IconComponent as React.ElementType}
        // a11y mui issue: combination "component + titleAccess" does not work: https://github.com/mui/material-ui/issues/33421
        titleAccess={title}
        {...props}
      />
    </div>
  );
};

export default Icon;
