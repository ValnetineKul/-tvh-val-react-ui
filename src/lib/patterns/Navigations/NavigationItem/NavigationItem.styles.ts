import { makeStyles } from '../../../../themes/core';

const useStyles = makeStyles<void, 'selected' | 'horizontalButtonBase'>({ name: 'App' })((theme, _, classes) => ({
  horizontalItem: {
    '&[role=menuitem]': {
      display: 'inline-flex',
    },

    [`& .${classes.horizontalButtonBase}`]: {
      minWidth: 'auto',
      minHeight: 'auto',
      padding: `${theme.layoutSpacing['spacing/300']} ${theme.layoutSpacing['spacing/400']}`,
      '&>*': {
        marginRight: theme.layoutSpacing['spacing/000'],
      },

      flexDirection: 'column',
      alignItems: 'center',
      '&:focus': {
        background: 'transparent',
      },
    },

    [`&.${classes.selected}`]: {
      '&:before': {
        content: '""',
        position: 'absolute',
        left: 0,
        bottom: 0,
        width: '100%',
        height: 2,
        background: theme.color['border/onSurfaceCurrent/selected'],
        zIndex: 1,
      },
    },
  },

  verticalItem: {
    position: 'relative',

    [`&.${classes.selected}`]: {
      '&:before': {
        content: '""',
        position: 'absolute',
        left: 0,
        top: 0,
        height: '100%',
        width: 4,
        background: theme.color['border/onSurfaceCurrent/selected'],
        zIndex: 1,
      },
    },
  },

  selected: {},
  horizontalButtonBase: {},
}));

export default useStyles;
