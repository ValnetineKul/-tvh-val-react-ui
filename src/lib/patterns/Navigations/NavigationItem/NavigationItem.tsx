import type { FC } from 'react';
import React from 'react';
import { useSurface } from '../../../../themes/core';
import { useButtonBase } from '../../../components/ButtonBase';
import MenuItem from '../../../components/Menus/MenuItem';
import useStyles from './NavigationItem.styles';
import type { NavigationItemProps } from './NavigationItem.types';

const NavigationItem: FC<NavigationItemProps> = ({
  selected = false,
  className,
  onClick,
  itemProps,
  label,
  horizontal: isHorizontal,
  icon,
  vertical: isVertical,
  startIcon,
  endIcon,
  subLabel,
  tag,
  ...props
}) => {
  const { classes, cx } = useStyles();
  const buttonBaseProps = useButtonBase(props);
  const { color: onSurface } = useSurface();

  const commonProps = {
    ...buttonBaseProps,
    selected,
    color: onSurface,
    onClick,
    buttonProps: itemProps,
  };

  if (isHorizontal) {
    const commonHorizontalProps = {
      ...commonProps,
      className: cx({ [classes.horizontalItem]: isHorizontal, [classes.selected]: selected }, className),
      buttonBaseClassName: classes.horizontalButtonBase,
    };

    return (
      <>
        {!icon && label && <MenuItem {...commonHorizontalProps} label={label} />}
        {icon && !label && <MenuItem {...commonHorizontalProps} label="" startIcon={icon} />}
        {icon && label && <MenuItem {...commonHorizontalProps} label={label} startIcon={icon} />}
      </>
    );
  }

  if (isVertical) {
    const commonVerticalProps = {
      ...commonProps,
      className: cx({ [classes.verticalItem]: isVertical, [classes.selected]: selected }, className),
    };

    return (
      <>
        {!startIcon && !endIcon && !subLabel && label && <MenuItem {...commonVerticalProps} label={label} />}
        {startIcon && label && !endIcon && !subLabel && (
          <MenuItem {...commonVerticalProps} label={label} startIcon={startIcon} />
        )}
        {endIcon && label && !startIcon && <MenuItem {...commonVerticalProps} label={label} endIcon={endIcon} />}
        {startIcon && endIcon && label && (
          <MenuItem {...commonVerticalProps} label={label} startIcon={startIcon} endIcon={endIcon} />
        )}
        {subLabel && label && !startIcon && !tag && (
          <MenuItem {...commonVerticalProps} label={label} subLabel={subLabel} />
        )}
        {subLabel && label && startIcon && tag && (
          <MenuItem {...commonVerticalProps} label={label} tag={tag} subLabel={subLabel} startIcon={startIcon} />
        )}
      </>
    );
  }

  return null;
};

export default NavigationItem;
