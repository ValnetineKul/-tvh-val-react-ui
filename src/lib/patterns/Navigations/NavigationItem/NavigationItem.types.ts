import type { ElementType, MouseEventHandler, ReactElement, ReactNode } from 'react';

import type { ButtonBaseProps } from '../../../components/ButtonBase';
import type { DataAttributes } from '../../../types/common';

type Never<T> = { [P in keyof T]?: never };

type CommonProps = ButtonBaseProps & {
  selected?: boolean;
  className?: string;
  elementType?: ElementType;
  onClick?: MouseEventHandler<HTMLButtonElement | HTMLAnchorElement>;
  itemProps?: React.HTMLAttributes<HTMLButtonElement | HTMLAnchorElement> & DataAttributes;
};

type Label = {
  label: string | JSX.Element;
};

type StartIcon = {
  startIcon: ReactElement;
};

type EndIcon = {
  endIcon: ReactElement;
};

type Icon = {
  icon: ReactElement;
};

type SubLabel = {
  subLabel: ReactNode;
};

type Tag = {
  tag: ReactElement;
};

type ConfigurableVerticalProps = StartIcon & EndIcon & SubLabel & Tag;

type IgnoreConfigurableVerticalParams<K extends keyof ConfigurableVerticalProps> = Never<
  Omit<ConfigurableVerticalProps, K>
>;

type OnlyLabelVertical = Label & Never<ConfigurableVerticalProps>;
type OnlyStartIcon = StartIcon & IgnoreConfigurableVerticalParams<keyof StartIcon>;
type OnlyEndIcon = EndIcon & IgnoreConfigurableVerticalParams<keyof EndIcon>;
type OnlyStartIconAndEndIcon = StartIcon & EndIcon & IgnoreConfigurableVerticalParams<keyof (StartIcon & EndIcon)>;
type OnlySubLabel = SubLabel & IgnoreConfigurableVerticalParams<keyof SubLabel>;
type OnlyStartIconAndSubLabelAndTag = StartIcon &
  SubLabel &
  Tag &
  IgnoreConfigurableVerticalParams<keyof (StartIcon & SubLabel & Tag)>;

type OnlyLabelHorizontal = Label & Never<Icon>;
type OnlyIcon = Icon & Never<Label>;
type OnlyLabelAndIcon = Label & Icon;

type VerticalProps = {
  vertical: true;
} & (
  | OnlyLabelVertical
  | (Label & (OnlyStartIcon | OnlyEndIcon | OnlyStartIconAndEndIcon | OnlySubLabel | OnlyStartIconAndSubLabelAndTag))
);

type HorizontalProps = {
  horizontal: true;
} & (OnlyLabelHorizontal | OnlyIcon | OnlyLabelAndIcon);

type OnlyVerticalProps = VerticalProps & Never<Omit<HorizontalProps, keyof Label>>;
type OnlyHorizontalProps = HorizontalProps & Never<Omit<VerticalProps, keyof Label>>;

export type NavigationItemProps = CommonProps & (OnlyVerticalProps | OnlyHorizontalProps);
