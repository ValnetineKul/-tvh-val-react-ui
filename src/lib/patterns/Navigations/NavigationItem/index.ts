import NavigationItem from './NavigationItem';
import type { NavigationItemProps } from './NavigationItem.types';

export type { NavigationItemProps };
export default NavigationItem;
