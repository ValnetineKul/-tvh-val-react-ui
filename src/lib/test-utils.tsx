import userEvent from '@testing-library/user-event';
import { fireEvent, render } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { theme } from '../themes/tvh';
import { createTheme, ThemeProvider, SurfaceProvider } from '../themes/core';

export const muiTheme = createTheme(theme);

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MemoryRouter>
      <ThemeProvider theme={muiTheme}>
        <SurfaceProvider color="100">{children}</SurfaceProvider>
      </ThemeProvider>
    </MemoryRouter>
  );
};

const customRender = (component: React.ReactElement, options?: Parameters<typeof render>[1]) =>
  render(component, { wrapper: AllTheProviders, ...options });

const customRenderHook: typeof renderHook = (hook, options) =>
  renderHook(hook, { wrapper: AllTheProviders, ...options });

export const initResponsiveTest = (size: number | 'Mobile' | 'Tablet' | 'Desktop' | 'LgDesktop' | 'XlDesktop') => {
  let width: number;
  if (size === 'Mobile') width = muiTheme.breakpoints.values.sm - 1;
  else if (size === 'Tablet') width = muiTheme.breakpoints.values.sm;
  else if (size === 'Desktop') width = muiTheme.breakpoints.values.md;
  else if (size === 'LgDesktop') width = muiTheme.breakpoints.values.lg;
  else if (size === 'XlDesktop') width = muiTheme.breakpoints.values.xl;
  else width = size;

  Object.defineProperty(window, 'innerWidth', {
    writable: true,
    configurable: true,
    value: width,
  });

  window.matchMedia = jest.fn().mockImplementation((query) => {
    const queryWidthInterval: string[] = query.match(/\d+(\.\d+)?/g);

    let matches;
    if (queryWidthInterval.length === 2) {
      matches = width >= +queryWidthInterval[0] && width < +queryWidthInterval[1];
    } else {
      const queryWidth = +queryWidthInterval[0];
      const isDown = /max/.test(query);

      matches = isDown ? width <= queryWidth : width >= queryWidth;
    }

    return {
      matches,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
    };
  });
};

const customUserEvent = Object.assign(userEvent, {
  mouseDown: fireEvent.mouseDown,
  mouseUp: fireEvent.mouseUp,
  mouseMove: fireEvent.mouseMove,
  scroll: fireEvent.scroll,
  error: fireEvent.error,
});

export { screen, within, waitFor, act, waitForElementToBeRemoved } from '@testing-library/react';
// override render method
export { customRender as render, customRenderHook as renderHook, customUserEvent as userEvent };
