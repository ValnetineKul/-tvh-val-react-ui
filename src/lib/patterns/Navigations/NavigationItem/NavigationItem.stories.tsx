import React from 'react';
import { Link } from 'react-router-dom';
import NavigationItem from './NavigationItem';
import { UserCircle, AngleRight } from '../../../components/Icon/icons/functional';
import Icon from '../../../components/Icon';
import Tag from '../../../components/Tags/Tag';
import Typography from '../../../components/Typography';
import { createMeta, createTemplate } from '../../../story-utils';

const Template = createTemplate(NavigationItem);

const HorizontalPrimitive = Template.bind({});
HorizontalPrimitive.args = {
  horizontal: true,
  selected: false,
  component: Link,
  to: '/whereToGoTo',
  label: '',
};

export const HorizontalNone = Template.bind({});
HorizontalNone.args = {
  ...HorizontalPrimitive.args,
  label: 'Label',
};

HorizontalNone.argTypes = {
  horizontal: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

HorizontalNone.storyName = 'HorizontalNone';

export const HorizontalIconOnly = Template.bind({});
HorizontalIconOnly.args = {
  ...HorizontalPrimitive.args,
  icon: <Icon icon={UserCircle} />,
};

HorizontalIconOnly.argTypes = {
  label: {
    table: {
      disable: true,
    },
  },
  icon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  horizontal: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

HorizontalIconOnly.storyName = 'HorizontalIconOnly';

export const HorizontalTopIcon = Template.bind({});
HorizontalTopIcon.args = {
  ...HorizontalPrimitive.args,
  icon: <Icon icon={UserCircle} />,
  label: 'Label',
};

HorizontalTopIcon.argTypes = {
  icon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  horizontal: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

HorizontalTopIcon.storyName = 'HorizontalTopIcon';

const VerticalPrimitive = Template.bind({});

VerticalPrimitive.args = {
  vertical: true,
  selected: false,
  label: '',
};

export const VerticalText = Template.bind({});
VerticalText.args = {
  ...VerticalPrimitive.args,
  label: 'Label',
};

VerticalText.argTypes = {
  vertical: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

VerticalText.storyName = 'VerticalText';

export const VerticalStartIcon = Template.bind({});
VerticalStartIcon.args = {
  ...VerticalPrimitive.args,
  startIcon: <Icon icon={UserCircle} />,
  label: 'Label',
};

VerticalStartIcon.argTypes = {
  startIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  vertical: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

VerticalStartIcon.storyName = 'VerticalStartIcon';

export const VerticalEndIcon = Template.bind({});
VerticalEndIcon.args = {
  ...VerticalPrimitive.args,
  endIcon: <Icon icon={AngleRight} />,
  label: 'Label',
};

VerticalEndIcon.argTypes = {
  endIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  vertical: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

VerticalEndIcon.storyName = 'VerticalEndIcon';

export const VerticalStartAndEndIcon = Template.bind({});
VerticalStartAndEndIcon.args = {
  ...VerticalPrimitive.args,
  startIcon: <Icon icon={UserCircle} />,
  endIcon: <Icon icon={AngleRight} />,
  label: 'Label',
};

VerticalStartAndEndIcon.argTypes = {
  startIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  endIcon: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
  vertical: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

VerticalStartAndEndIcon.storyName = 'VerticalStartAndEndIcon';

export const VerticalInfoBottom = Template.bind({});
VerticalInfoBottom.args = {
  ...VerticalPrimitive.args,
  label: 'Label',
  subLabel: 'subLabel',
};

VerticalInfoBottom.argTypes = {
  subLabel: {
    table: {
      disable: false,
    },
  },
  vertical: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

VerticalInfoBottom.storyName = 'VerticalInfoBottom';

export const VerticalTagAndSubLabelAndStartIcon = Template.bind({});
VerticalTagAndSubLabelAndStartIcon.args = {
  ...VerticalPrimitive.args,
  label: 'Label',
  tag: <Tag label="Tag" />,
  startIcon: <Icon icon={UserCircle} />,
  subLabel: (
    <Typography component="div" variant="body400" secondary>
      SubLabel Component
    </Typography>
  ),
};

VerticalTagAndSubLabelAndStartIcon.argTypes = {
  subLabel: {
    table: {
      disable: false,
    },
    control: { disable: true },
  },
  tag: {
    table: {
      disable: false,
    },
    control: { disable: true },
  },
  vertical: {
    table: {
      disable: false,
    },
    control: {
      disable: true,
    },
  },
};

VerticalTagAndSubLabelAndStartIcon.storyName = 'VerticalTagAndSubLabelAndStartIcon';

export default createMeta({
  component: NavigationItem,
  title: 'Patterns/Navigations/NavigationItem',
  argTypes: {
    onClick: {
      action: { argTypesRegex: '^on.*' },
      control: { disable: true },
    },
    disabled: {
      table: {
        disable: true,
      },
    },
    icon: {
      table: {
        disable: true,
      },
    },
    subLabel: {
      table: {
        disable: true,
      },
    },
    tag: {
      table: {
        disable: true,
      },
    },
    startIcon: {
      table: {
        disable: true,
      },
    },
    endIcon: {
      table: {
        disable: true,
      },
    },
    horizontal: {
      table: {
        disable: true,
      },
    },
    vertical: {
      table: {
        disable: true,
      },
    },
    elementType: {
      control: {
        disable: true,
      },
      table: {
        type: { detail: 'React or HTML element type' },
      },
    },
    component: {
      control: {
        disable: true,
      },
    },
    href: {
      control: {
        disable: true,
      },
      table: {
        type: { detail: 'If "href" is filled, the component will immediately transform to HTML link tag "a"' },
      },
    },
    to: {
      control: {
        disable: true,
      },
    },
    download: {
      control: {
        disable: true,
      },
      table: {
        type: {
          detail:
            'If "href" and "download" are filled â€“ downloads the file when clicking on link (instead of navigating to the file)',
        },
      },
    },
  },
});
