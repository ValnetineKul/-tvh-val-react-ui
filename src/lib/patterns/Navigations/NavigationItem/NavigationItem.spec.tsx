import React from 'react';
import { Link } from 'react-router-dom';
import { render, screen } from '../../../test-utils';
import NavigationItem from './NavigationItem';
import { UserCircle, AngleRight } from '../../../components/Icon/icons/functional';
import Icon from '../../../components/Icon';

describe('NavigationItem', () => {
  describe('Horizontal', () => {
    it('Should render correctly horizontal', () => {
      const { container } = render(
        <NavigationItem icon={<Icon icon={UserCircle} />} horizontal component={Link} to="/" label="Label" />
      );
      expect(container.firstChild).toMatchSnapshot('horizontal');
    });
    it('Should be selected', () => {
      render(<NavigationItem horizontal component={Link} to="/" label="Label" selected />);
      const item = screen.getByRole('menuitem');
      expect(item.className).toMatch('selected');
    });
    it('Should render icon', () => {
      render(
        <NavigationItem
          horizontal
          icon={<Icon icon={UserCircle} title="icon" />}
          component={Link}
          to="/"
          label="Label"
        />
      );
      expect(screen.getByText(/userCircle.svg/)).toBeInTheDocument();
    });
  });

  describe('Vertical', () => {
    it('Should render correctly vertical', () => {
      const { container } = render(<NavigationItem vertical component={Link} to="/" label="Label" />);
      expect(container.firstChild).toMatchSnapshot('vertical');
    });
    it('Should be selected', () => {
      render(<NavigationItem vertical component={Link} to="/" label="Label" selected />);
      const item = screen.getByRole('menuitem');
      expect(item.className).toMatch('selected');
    });
    it('Should render startIcon', () => {
      render(
        <NavigationItem
          vertical
          startIcon={<Icon icon={UserCircle} title="startIcon" />}
          component={Link}
          to="/"
          label="Label"
        />
      );
      expect(screen.getByText(/userCircle.svg/)).toBeInTheDocument();
    });
    it('Should render endIcon', () => {
      render(
        <NavigationItem
          vertical
          endIcon={<Icon icon={AngleRight} title="endIcon" />}
          component={Link}
          to="/"
          label="Label"
        />
      );
      expect(screen.getByText(/angleRight.svg/)).toBeInTheDocument();
    });
    it('Should render startIcon and endIcon', () => {
      render(
        <NavigationItem
          vertical
          startIcon={<Icon icon={UserCircle} title="startIcon" />}
          endIcon={<Icon icon={AngleRight} title="endIcon" />}
          component={Link}
          to="/"
          label="Label"
        />
      );
      expect(screen.getByText(/userCircle.svg/)).toBeInTheDocument();
      expect(screen.getByText(/angleRight.svg/)).toBeInTheDocument();
    });
    it('Should render subLabel', () => {
      render(<NavigationItem vertical to="/" label="Label" subLabel="Sub label" />);
      expect(screen.getByText(/Sub label/)).toBeInTheDocument();
    });
  });
});
