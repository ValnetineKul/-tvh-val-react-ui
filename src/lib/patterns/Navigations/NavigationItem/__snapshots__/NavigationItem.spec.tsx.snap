// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationItem Horizontal Should render correctly horizontal: horizontal 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root tss-q6zrpx-App-horizontalItem mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-7-App-horizontalButtonBase-ref tss-1ie5n30-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper-App-horizontalButtonBase mui-10d1a0h-MuiButtonBase-root"
    href="/"
    tabindex="0"
  >
    <div
      class="tss-6-MenuItem-startIcon-ref tss-2hk6gu-Icon-root-Icon-sizeMd-MenuItem-startIcon"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium mui-i4bv87-MuiSvgIcon-root"
        color="inherit"
        focusable="false"
        viewBox="0 0 24 24"
      >
        userCircle.svg
      </svg>
    </div>
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Label
        </span>
      </div>
    </div>
  </a>
</li>
`;

exports[`NavigationItem Vertical Should render correctly vertical: vertical 1`] = `
<li
  class="tss-1qfvypo-MenuItem-root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root tss-1usksdx-App-verticalItem mui-t7a2rv-MuiButtonBase-root-MuiMenuItem-root"
  role="menuitem"
  tabindex="-1"
>
  <a
    aria-disabled="false"
    class="MuiButtonBase-root tss-6-MenuItem-wrapper-ref tss-19jpft6-ButtonBase-root-ButtonBase-focusInset-MenuItem-wrapper mui-10d1a0h-MuiButtonBase-root"
    href="/"
    tabindex="0"
  >
    <div
      class="tss-sqs6xo-MenuItem-labelWrapper-MenuItem-label"
    >
      <div
        class="tss-0-Typography-weightRegular-ref tss-10hdle-Typography-root-Typography-weightRegular-Tooltip-wrapper-TruncatedText-truncatedText-MenuItem-label-Typography-body400"
        data-mui-internal-clone-element="true"
      >
        <span
          class="tss-0-Typography-weightRegular-ref tss-1hhfpop-Typography-root-Typography-weightRegular-TruncatedText-truncatedText-Typography-body500"
        >
          Label
        </span>
      </div>
    </div>
  </a>
</li>
`;
